### PR TITLE
feat(Purchase registry) add link to stock inventory

### LINKED ISSUE #4417: Link purchase orders from "Articles in Stock" Registry
It would be nice to add a link to purchase orders for stock items from the the articles in stock registry.  It would show up right here:
image

Whoever tackles this could also tackle https://github.com/IMA-WorldHealth/bhima/issues/4305 at the same time.

The link should be to either:
  1. The purchase order that created this lot (if possible) OR
  2. Filter the purchase orders on the inventory item of this lot.

### PR DESCRIPTION
**Pull request by jeremielodi**
closes  #4417

### CODE REVIEW

File: client/src/modules/stock/inventories/registry.js [hunk 1]
```diff
@@ -212,6 +212,9 @@ function StockInventoriesController(
     vm.hasError = false;
     vm.loading = true;
 
+    if (filters.purchase_uuid) {
+      filters.includeEmptyLot = 1;
```
Comment (jniles) [inline client/src/modules/stock/inventories/registry.js]: Can we just send this from the client side instead of putting it on the server?

**Review by jniles — COMMENTED** [review]
This is actually the opposite of https://github.com/IMA-WorldHealth/bhima/issues/4417.  That issue was trying to go from stock -> purchase orders, but this PR goes from purchase -> stock.  It's still a good PR though.  It just doesn't close the issue.

File: client/src/modules/stock/inventories/registry.js (hunk 1 above)
Comment (jeremielodi) [reply client/src/modules/stock/inventories/registry.js]: okay

File: client/src/modules/purchases/templates/action.cell.html [hunk 2]
```diff
@@ -28,6 +28,16 @@
         <i class="fa fa-arrows-h"></i> <span translate>PATIENT_REGISTRY.STOCK_MOVEMENTS</span>
       </a>
     </li>
+    
+    <li>
+      <a data-method="view-stock-lot" href
+        ui-sref="stockInventories({ filters : [
+          { key : 'period', value : 'allTime' },
+          { key : 'includeEmptyLot', value : 0 },
```
Comment (jniles) [inline client/src/modules/purchases/templates/action.cell.html]: I think this should be `value : 1`.

File: server/controllers/stock/core.js [hunk 3]
```diff
@@ -585,7 +585,9 @@ function getInventoryQuantityAndConsumption(params, monthAverageConsumption, ena
     _status = params.status;
     delete params.status;
   }
-
+  if (params.purchase_uuid) {
```
Comment (jniles) [inline server/controllers/stock/core.js]: Could you remove this from the server?

**Review by jniles — COMMENTED** [review]
@jeremielodi, looks like you got half-way there.  Just a few more changes.

Also, could you remove the "closes" from the commit?  It won't close that issue.  You can do it with `git commit --amend`.

File: client/src/modules/purchases/templates/action.cell.html (hunk 2 above)
Comment (jeremielodi) [reply client/src/modules/purchases/templates/action.cell.html]: okay

**Review by jniles — APPROVED** [review]
LGTM.

bors r+

**Pull request merged**